docs/process/style_guide.md

```diff
@@ -602,6 +602,22 @@ fn main() {
 
 ## Data Types
 
+### Zero-copy in DataProvider structs :: required
+
+All data structs that can be passed through the DataProvider pipeline must support *zero-copy deserialization:* in practice, no heap allocations should be required when deserializing from Bincode-like formats. This means that if the type involves variable-length data like strings, vectors, and maps, it must use a zero-copy type backed by a byte buffer to represent them.
```

Comment:
I think we should also note that it should be a type that can also be fully owned -- so it should not have `&str` or `&T`, only cows and zerovecs